Sources/ComposableArchitecture/SwiftUI/ForEachStore.swift

```diff
@@ -78,64 +149,69 @@ import SwiftUI
 /// ```
 ///
 public struct ForEachStore<
-  EachState, EachAction, Data: Collection, ID: Hashable, Content: View
->: DynamicViewContent {
-  public let data: Data
-  let content: Content
+  StatesCollection: IdentifiedStatesCollection, EachAction, EachContent: View
+>: DynamicViewContent
+where StatesCollection.ID: Hashable {
+  public typealias EachState = StatesCollection.State
+  public typealias ID = StatesCollection.ID
+
+  let store: Store<StatesCollection, (ID, EachAction)>
+  @ObservedObject var viewStore: ViewStore<StatesCollection, (ID, EachAction)>
```

Comment:
Please note, a `ViewStore<StateCollection, _>` and not `ViewStore<StateCollection.IDs, _>` like before.